### PR TITLE
GUVNOR-3511: Use the same Delete popup for all column types

### PR DESCRIPTION
**Pull request by jomarko**
@karreiro @manstis I will be happy if you have a look. The result shown in the video below:
https://webmshare.com/NKLrx

### CODE REVIEW

**Comment by jomarko** [comment]
@karreiro great idea! I will definitely try to apply, tomorrow, Today I need a rest :)

**Comment by manstis** [comment]
@karreiro @jomarko IMO making ALL delete popups consistent is beyond the scope of this PR (I bet we still use ```Window.confirm(..)``` in some places for sure); but if you want I'm not going to grumble.

**Comment by jomarko** [comment]
@manstis please lets keep this open.

**Comment by jomarko** [comment]
@manstis I think @karreiro meant just popups for columns delete, what could be no such huge change ... ?

**Comment by jomarko** [comment]
@manstis @karreiro , Uf, it was not so easy as I thought, see the video below and say it satisfies your expectations, please please please! :D
https://webmshare.com/R3v1G

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java [hunk 1]
```diff
@@ -16,25 +16,34 @@
...
         setText(GuidedDecisionTableConstants.INSTANCE.Delete());
         setTitle(GuidedDecisionTableConstants.INSTANCE.DeleteThisColumn());
         addClickHandler((clickEvent) -> {
-            final String confirmMessage = GuidedDecisionTableConstants.INSTANCE.DeleteColumnWarning(columnHeader);
-            YesNoCancelPopup.newYesNoCancelPopup(GuidedDecisionTableConstants.INSTANCE.Delete(),
-                                                 confirmMessage,
-                                                 () -> afterDeleteConfirmedCommand.execute(),
-                                                 null,
-                                                 null).show();
+            deletePopUpPresenter.getToggleCommentPresenter().getViewElement().setAttribute("hidden",
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java]: @jomarko I don't think it's going to be quite so easy :-( IMO this is really an abuse of the ```DeletePopupPresenter``` as we should not be tweaking it's visual appearance from outside the class (adding a ```enableComment(boolean)``` is probably quite acceptable)... but that also brings me onto...

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java [hunk 2]
```diff
@@ -16,25 +16,34 @@
...
+                                                                                           "true");
+            deletePopUpPresenter.show((comment) -> afterDeleteConfirmedCommand.execute());
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java]: The ```DeletePopupPresenter.View``` shows a "Are you sure you want to delete this **Asset**" prompt. We'd need to be able to make this "Are you sure you want to delete this **column**?". There's probably different ways you might be able to do this; either investigate a ```DeletePopupPresenter.setPrompt(String)``` method (that calls ```View.setPrompt(String)``` that calls ```modal.setTitle( prompt )```) or have your own implementation of ```DeletePopupPresenter``` that injects a different ```View``` that sets the caption to "Are you sure you want to delete this **column**?" but that's a lot of leg work). Up to you.

**Comment by jomarko** [comment]
Ensemble with AppFormer/uberfire#843

**Comment by jomarko** [comment]
@manstis  Please give me a second chance https://webmshare.com/9rN3y I linked also changes in the uberfire repo.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java [hunk 3]
```diff
@@ -133,10 +132,18 @@ public void execute() {
...
     private final RestrictedMousePanMediator mousePanMediator = restrictedMousePanMediator();
 
+    @Inject
     private ColumnManagementView actionsPanel;
 
+    @Inject
     private ColumnManagementView conditionsPanel;
 
+    @Inject
+    private Instance<AttributeColumnConfigRow> attributeColumnConfigRows;
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java]: Sorry.. I missed this last time around.. best to make this ```ManagedInstance``` to ensure instances are released.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java [hunk 4]
```diff
@@ -133,10 +132,18 @@ public void execute() {
...
+
+    @Inject
+    private Instance<DeleteColumnManagementAnchorWidget> deleteColumnManagementAnchorWidgets;
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java]: Sorry.. I missed this last time around.. best to make this ```ManagedInstance``` to ensure instances are released.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/ColumnManagementView.java [hunk 5]
```diff
@@ -36,15 +39,24 @@
...
 import org.drools.workbench.screens.guided.dtable.client.widget.table.GuidedDecisionTableModellerView;
 import org.drools.workbench.screens.guided.dtable.client.widget.table.utilities.ColumnUtilities;
-import org.uberfire.ext.widgets.common.client.common.popups.YesNoCancelPopup;
 import org.uberfire.ext.widgets.common.client.common.popups.errors.ErrorPopup;
-import org.uberfire.mvp.ParameterizedCommand;
 
+@Dependent
 public class ColumnManagementView extends VerticalPanel {
 
+    private Instance<DeleteColumnManagementAnchorWidget> deleteColumnManagementAnchorWidgets;
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/ColumnManagementView.java]: Sorry.. I missed this last time around.. best to make this ```ManagedInstance``` to ensure instances are released.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Inject
+    private DeletePopUpPresenter deletePopUpPresenter;
+
+    public DeleteColumnManagementAnchorWidget() {
+        super();
+        setText(GuidedDecisionTableConstants.INSTANCE.Delete());
+        setTitle(GuidedDecisionTableConstants.INSTANCE.DeleteThisColumn());
+        addClickHandler((clickEvent) -> {
+            deletePopUpPresenter.setPrompt(GuidedDecisionTableConstants.INSTANCE.DeleteColumnWarning(columnHeader));
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java]: Wouldn't it be nice to have a test verifying you're hiding the comments box and changing the prompt?

**Comment by jomarko** [comment]
@manstis fixes in new commit. Let me know if you find something else I could improve.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by karreiro** [comment]
( @jomarko the tests are broken because this PR depends on https://github.com/AppFormer/uberfire/pull/843, but the name of the branches are different.. so, Jenkins doesn't know that the UF PR should be built before)

**Comment by jomarko** [comment]
@karreiro yes, Mike have told me already, recreating uberfire PR right now.

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by ederign** [comment]
jenkins please retest this

**Pull request merged**